explorer/interpreter/interpreter.cpp

```diff
@@ -1087,6 +1087,56 @@ auto Interpreter::CallFunction(const CallExpression& call,
   }
 }
 
+// Returns true if the format string is okay to pass to formatv. This only
+// supports `{{` and `{N}` as special syntax.
+static auto IsValidFormatString(const char* format_string, int num_args)
+    -> bool {
+  const char* cursor = format_string;
+  while (true) {
+    switch (*cursor) {
+      case '\0':
+        // End of string.
+        return true;
+      case '{':
+        // `{` is a special character.
+        ++cursor;
+        switch (*cursor) {
+          case '\0':
+            // Early end.
+            return false;
+          case '{':
+            // Escaped `{`.
+            ++cursor;
+            break;
+          case '}':
+            // Reject `{}`.
+            return false;
+          default:
+            int index = 0;
+            while (*cursor != '}') {
+              if (*cursor < '0' || *cursor > '9') {
+                // Non-numeric text, including `\0`.
+                return false;
+              }
+              index = (10 * index) + (*cursor - '0');
```

Comment:
doesn't handle overflow.

```
Print("{2147483648}", 1);
```
weirdly prints `2147483648`

Reply:
Blocked and added a test.